jdaviz/core/template_mixin.py

```diff
@@ -1899,22 +1904,32 @@ def add_results_from_plugin(self, data_item, replace=None, label=None):
             # entry should be the same as the original entry (to avoid deleting reference data)
             add_to_viewer_refs = []
             add_to_viewer_vis = []
+            preserved_attributes = []
             for viewer_select_item in self.add_to_viewer_items[1:]:
                 # index 0 is for "None"
                 viewer_ref = viewer_select_item['reference']
                 viewer_item = self.app._viewer_item_by_reference(viewer_ref)
                 viewer = self.app.get_viewer(viewer_ref)
-                viewer_loaded_labels = [layer.layer.label for layer in viewer.layers]
-                if label in viewer_loaded_labels:
-                    add_to_viewer_refs.append(viewer_ref)
-                    add_to_viewer_vis.append(label in viewer_item['visible_layers'])
+                for layer in viewer.layers:
+                    if layer.layer.label != label:
+                        continue
+                    else:
+                        add_to_viewer_refs.append(viewer_ref)
+                        add_to_viewer_vis.append(label in viewer_item['visible_layers'])
```

Comment:
couldn't this result in duplicating the entry in the list if there are multiple layers?  (I'm trying to think of what the consequences would be to test...)

Reply:
Hmm, perhaps, you're right that this logic isn't perfectly equivalent to the previous one. How about just adding a `break` here in the else clause?

Reply:
I guess since you're only acting on the _current_ viewer, only one layer should ever enter the `else`, so its probably fine (which explains why I couldn't find a way to break it in testing).  It might be more obvious for future us though to do something like 

```
layer = next((layer for layer in viewer.layers if layer.layer.label == label), None)
# everything that is currently in the else
```